nc_test4/tst_xplatform2.c

```diff
@@ -209,16 +210,16 @@ check_file_3(int ncid, struct s3 *data_out)
    for (i = 0; i < DIM3_LEN; i++)
       for (j = 0; j < NUM_VL; j++)
       {
-	 if (data_in[i].data[j].len != data_in[i].data[j].len) ERR;
+	 if (data_in[i].data[j].len != data_out[i].data[j].len) ERR;
```

Comment:
This is an important warning fix. We were comparing data_in to data_in, which of course was always equal. We need to compare data_in to data_out to actually test results of read.